pkg/readiness/ready_tracker.go

```diff
@@ -853,6 +887,7 @@ func (t *Tracker) trackData(ctx context.Context, gvk schema.GroupVersionKind, dt
 	lister := retryLister(t.lister, retryUnlessUnregistered)
 	err := lister.List(ctx, u)
 	if err != nil {
+		hadError = true
```

Comment:
Same here, should I return nil and remove the hadError check flag cause its fail open?